src/transaction/deferred_action_manager.cpp

```diff
@@ -0,0 +1,35 @@
+#include "transaction/deferred_action_manager.h"
+
+#include "common/macros.h"
+
+namespace terrier::transaction {
+
+// TODO(John:GC) Where should this live?
```

Comment:
Ideally, this remains solely in the existing garbage collector files with the goal of ultimately removing this logic by handling it in the index wrappers through deferrals.